src/sentry/snuba/tasks.py

```diff
@@ -184,7 +184,16 @@ def delete_subscription_from_snuba(query_subscription_id, **kwargs):
         )
 
     if subscription.status == QuerySubscription.Status.DELETING.value:
+        snuba_query_id = subscription.snuba_query.id
```

Comment:
You could do `snuba_query = subscription.snuba_query` here and skip the fetch after, since this line is already making the query. Alternatively, `snuba_query_id = subscription.snuba_query_id`